counterexamples/char_p_zero_ne_char_zero.lean

```diff
@@ -0,0 +1,25 @@
+/-
+Copyright (c) 2022 Damiano Testa. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Damiano Testa, Eric Wieser
+-/
+import algebra.char_p.basic
+
+/-! # `char_p R 0` and `char_zero R` need not coincide for semirings
+
+This file shows that there are semiring `R` for which `char_p R 0` holds and `char_zero R` does not.
+
+The example is `{0, 1}` with saturating addition.
```

Comment:
Do you think we should have a file in mathlib about this type? It's now used in two counterexamples.

Reply:
Maybe yes, or at least put the two counterexamples in the same file.

Reply:
Actually, I wonder whether it might be a better idea to create folders corresponding to the types used, when it makes sense.  Thus, we could have a folder `two` containing this file and `homogeneous_prime_not_prime.lean`.  Since the definitions and properties of the type are very short, it may not be needed now to have a separate file setting up all the instances, but maybe in the future it might make sense.

What do you think?

Reply:
Am I right in thinking that `with_zero unit` is the "boolean semifield"? Should we just add a new `boolean_semifield.lean` file somewhere?

Reply:
I think that indeed `with_zero unit` I the Boolean semifield, although I had to look it up to see what you meant!

We can certainly create a file with it.

By the way, I wanted to use your `dual_number ℤ` for an example re `sign_type` but failed to define a ring structure on it (or the lex order on its coefficients).

Anyway, I really like these files about explicit types!

Reply:
I'll give it a try when I find some time.  If you prefer to get started, go ahead!  I imagine my version would be changed to yours anyway!  ;-)